execute/dataset_test.go

```diff
@@ -0,0 +1,221 @@
+package execute_test
+
+import (
+	"testing"
+
+	arrowmem "github.com/apache/arrow/go/arrow/memory"
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/array"
+	"github.com/influxdata/flux/arrow"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/execute/executetest"
+	"github.com/influxdata/flux/execute/table"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/mock"
+)
+
+func TestTransportDataset_Process(t *testing.T) {
```

Comment:
I so so _so_ prefer these tests to the tests that create an array of inputs and outputs and iterates over them in a single function. I could read these like a narrative, which helped me get my head around what was actually being tested without scrolling back and forth.